types/qaengine/commonqa/commonqa.go

```diff
@@ -76,3 +79,45 @@ func IngressHost(defaulthost string) string {
 func MinimumReplicaCount(defaultminreplicas string) string {
 	return qaengine.FetchStringAnswer(common.ConfigMinReplicasKey, "Provide the minimum number of replicas each service should have", []string{"If the value is 0 pods won't be started by default"}, defaultminreplicas)
 }
+
+// GetPortsForService returns ports used by a service
+func GetPortsForService(detectedPorts []int32) []int32 {
+	var selectedPortsStr, enteredPortsStr, detectedPortsStr []string
+	var enteredPorts string
+	var exposePorts []int32
+	if len(detectedPorts) != 0 {
+		for _, detectedPort := range detectedPorts {
+			detectedPortsStr = append(detectedPortsStr, strconv.Itoa(int(detectedPort)))
+		}
+		allDetectedPortsStr := append(detectedPortsStr, qatypes.OtherAnswer)
+		selectedPortsStr = qaengine.FetchMultiSelectAnswer(common.ConfigServicesKey+common.Delim+common.ConfigPortsForServiceKeySegment, "Select ports to be exposed :", []string{"Select Other if you want to add more ports"}, detectedPortsStr, allDetectedPortsStr)
+	}
+	if len(selectedPortsStr) == 0 || common.IsStringPresent(selectedPortsStr, qatypes.OtherAnswer) {
+		enteredPorts = qaengine.FetchMultilineAnswer(common.ConfigServicesKey+common.Delim+common.ConfigAdditionalPortsForServiceKeySegment, "Enter the ports to be exposed", []string{"Enter each port in a newline"}, "")
+	}
+
+	enteredPortsStr = strings.Split(enteredPorts, "\n")
+	for _, portStr := range selectedPortsStr {
+		if portStr != qatypes.OtherAnswer {
+			if portStr != "" {
+				port, err := strconv.ParseInt(portStr, 10, 32)
+				if err != nil {
+					logrus.Errorf("Error while converting the selected port from string to int : %s", err)
+				} else {
+					exposePorts = append(exposePorts, int32(port))
+				}
+			}
+		}
+	}
+	for _, portStr := range enteredPortsStr {
+		if portStr != "" {
+			port, err := strconv.ParseInt(portStr, 10, 32)
+			if err != nil {
+				logrus.Errorf("Error while converting the entered port from string to int : %s", err)
+			} else {
+				exposePorts = append(exposePorts, int32(port))
```

Comment:
Check if the port already existing using common.IsIntPresent() before adding

Reply:
Yes sure, I was also thinking if the user re-enters some port which he already selected in previous question, then same port will get exposed twice. I will include that check. Thanks for your valuable feedback. :)